src/Storages/PostgreSQL/MaterializedPostgreSQLConsumer.cpp

```diff
@@ -507,47 +549,94 @@ void MaterializedPostgreSQLConsumer::processReplicationMessage(const char * repl
                 return;
             }
 
+            auto log_table_structure_changed = [&](std::string_view reason)
```

Comment:
I'm not entirely sure, but it seems like there might be a potential issue with the lifetime of the temporary std::string object returned by fmt::format. 

```suggestion
            auto log_table_structure_changed = [&](const std::string & reason)
```